packages/angular/src/migrations/update-16-0-0/remove-karma-defaults.ts

```diff
@@ -0,0 +1,77 @@
+import type {
+  NxJsonConfiguration,
+  ProjectConfiguration,
+  Tree,
+} from '@nx/devkit';
+import {
+  formatFiles,
+  getProjects,
+  readNxJson,
+  updateNxJson,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+
+const GENERATORS = ['application', 'library', 'host', 'remote'];
+
+export default async function removeKarmaDefaults(tree: Tree) {
+  const nxJson = readNxJson(tree);
+
+  if (nxJson.generators) {
+    const updatedConfig = updateUnitTestRunner(nxJson.generators);
+
+    if (updatedConfig) {
+      updateNxJson(tree, nxJson);
+    }
+  }
+
+  const projects = getProjects(tree);
+
+  for (const [projectName, projectConfig] of projects) {
+    if (projectConfig.generators) {
+      const updatedProject = updateUnitTestRunner(projectConfig.generators);
+
+      if (updatedProject) {
+        updateProjectConfiguration(tree, projectName, projectConfig);
+      }
+    }
+  }
+
+  await formatFiles(tree);
+}
+
+function updateUnitTestRunner(
+  generatorsConfig:
+    | NxJsonConfiguration['generators']
+    | ProjectConfiguration['generators']
+) {
+  const generators = Object.entries(generatorsConfig);
+
+  let updatedConfig = false;
+  for (const [generatorName, generatorDefaults] of generators) {
+    for (const possibleGenerator of GENERATORS) {
+      if (
+        generatorDefaults[possibleGenerator] &&
+        generatorDefaults[possibleGenerator]['unitTestRunner'] &&
+        generatorDefaults[possibleGenerator]['unitTestRunner'] === 'karma'
+      ) {
+        generatorsConfig[generatorName][possibleGenerator]['unitTestRunner'] =
+          undefined;
+        updatedConfig = true;
+      }
+    }
+
+    if (!GENERATORS.map((v) => `@nrwl/angular:${v}`).includes(generatorName)) {
```

Comment:
We must also account for `@nx/angular:<generator>`. A migration for a previous version renames them from `@nrwl` to `@nx`.